core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -170,8 +170,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         if (executorDataMap.contains(executorId)) {
           executorRef.send(RegisterExecutorFailed("Duplicate executor ID: " + executorId))
           context.reply(true)
-        } else if (scheduler.nodeBlacklist != null &&
-          scheduler.nodeBlacklist.contains(hostname)) {
+        } else if (scheduler.nodeBlacklistWithExpiryTimes.contains(hostname)) {
```

Comment:
this change seems to be causing all the test failures.  Its because there are tests that use a mock TaskSchedulerImpl, and so `scheduler.nodeBlacklistWithExpiryTimes` returns null from the mock.

Reply:
I like the current solution. As on production nodeBlacklist really cannot be null and that null came from the tests. So we adopted the tests to be more close to the production case (and not the other way around: adapting the production code to be prepared for the side effect of the testing).